### PR TITLE
python3Packages.pytest-filter-subpackage: move pytestCheckHook to checkInputs

### PR DESCRIPTION
**Pull request by applePrincess**
###### Motivation for this change
Fix unnecessarily propagating pytestCheckHook in python3Packages.pytest-filter-subpackage.
Add missing dependency in python3Packages.sunpy due to above.  

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 133497` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.spectral-cube</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytest-astropy/default.nix [hunk 1]
```diff
@@ -39,6 +39,7 @@ buildPythonPackage rec {
 
   # pytest-astropy is a meta package and has no tests
   doCheck = false;
+  dontUsePytestCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-astropy/default.nix:42]: This shouldn't be required and shows us that some dependency is propagating pytestCheckHook.

Comment (applePrincess) [reply pkgs/development/python-modules/pytest-astropy/default.nix:42]: I found one propagating it.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133497` at df31d830 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built successfully:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-filter-subpackage</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-filter-subpackage</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.spectral-cube</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/pytest-filter-subpackage/default.nix:38:3:
```
   |
38 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/sunpy/default.nix [hunk 2]
```diff
@@ -67,6 +68,7 @@ buildPythonPackage rec {
 
   checkInputs = [
     hypothesis
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sunpy/default.nix:71]: This should be pytestCheckHook.

Comment (applePrincess) [reply pkgs/development/python-modules/sunpy/default.nix]: fixed it!

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 133497` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-filter-subpackage</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-filter-subpackage</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.spectral-cube</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> Fix unnecessarily propagating pytestCheckHook in python3Packages.pytest-filter-subpackage.

This is actually a hard bug and causes weird errors down the line. Good that you found it.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 133497` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
16 packages built:
</summary>
<ul>
<li>python38Packages.aplpy</li>
<li>python38Packages.asdf</li>
<li>python38Packages.pyregion</li>
<li>python38Packages.pytest-astropy</li>
<li>python38Packages.pytest-filter-subpackage</li>
<li>python38Packages.reproject</li>
<li>python38Packages.spectral-cube</li>
<li>python38Packages.sunpy</li>
<li>python39Packages.aplpy</li>
<li>python39Packages.asdf</li>
<li>python39Packages.pyregion</li>
<li>python39Packages.pytest-astropy</li>
<li>python39Packages.pytest-filter-subpackage</li>
<li>python39Packages.reproject</li>
<li>python39Packages.spectral-cube</li>
<li>python39Packages.sunpy</li>
</ul>
</details>

**Pull request merged**